utils/src/main/java/software/amazon/awssdk/utils/ImmutableMap.java

```diff
@@ -199,7 +199,9 @@ public static <K, V> ImmutableMap<K, V> of(K k0, V v0, K k1, V v1,
     private static <K, V> void putAndWarnDuplicateKeys(Map<K, V> map, K key,
                                                        V value) {
         if (map.containsKey(key)) {
-            throw new IllegalArgumentException(DUPLICATED_KEY_MESSAGE);
+            log.error(() -> String.format("Duplicate keys are provided for [%s]. The first value [%s] will be kept, and the "
```

Comment:
This can result in backward incompatible change.
Instead of changing the Immutable Map which is generic, Sdkprotected and where it always expects user to supply the non duplicate keys can we add try catch specifically to the `SIGNING_REGIONS_BY_REGION` individual maps or just change it to another Map where it accepts duplicate keys

Reply:
may be something like 
https://github.com/google/guava/blob/5c23347cc162a1dcb046b034ba8db1db4f0de77b/android/guava/src/com/google/common/collect/ImmutableMap.java#L563-L565